buildlib/azure-pipelines.yml

```diff
@@ -10,33 +10,43 @@ resources:
     - container: centos7
       image: ucfconsort.azurecr.io/ucx/centos7:1
       endpoint: ucfconsort_registry
+    - container: fedora
+      image: ucfconsort.azurecr.io/ucx/fedora:30
+      endpoint: ucfconsort_registry
 
 stages:
-  - stage: Build
+  - stage: LatestCCs
     jobs:
-      - job: Compile
-        displayName: Compile Tests
-        pool:
-          vmImage: 'Ubuntu-16.04'
+      - job: latest_cc
+        displayName: Latest CCs and CppCheck
+        container: fedora
         steps:
-          - bash: |
-              ./autogen.sh
+          - bash: ./autogen.sh
             displayName: Setup autotools
 
-          - bash: |
-              mkdir build && cd build
-              ../configure --disable-numa
+          - bash: ./contrib/configure-release
             displayName: Configure
 
+          # this will also generate compile_commands.json for cppcheck
+          - bash: bear make -j`nproc` V=1
+            displayName: GCC
+
           - bash: |
-               cd build
-               gcc -v
-               make -s -j `nproc`
-            displayName: Build gcc 5.4
+              make clean
+              export CC=clang; export CXX=clang++
+              ./contrib/configure-release
+              make -j`nproc` V=1
+            displayName: Clang
 
+          - bash: |
+              cppcheck --project=compile_commands.json --template=gcc --error-exitcode=1 --inline-suppr >/dev/null
+            displayName: CppCheck
+
+  - stage: Build
```

Comment:
Yes. I don't what build matrix (it will grow) to run if build fails on the latest compilers.

Reply:
actually i think we can run it in parallel, to report more errors in one build. and in the common case when there are no errors, it will complete faster.

Reply:
OK, will move it under the existing stage.